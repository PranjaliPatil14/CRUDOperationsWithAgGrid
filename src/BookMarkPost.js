import React from "react";

const BookMarkPost = ({ api, data, node }) => {
  const markBookMark = () => {
    data.bookMark = !data.bookMark;
    api.applyTransaction({ update: [data] });
    api.refreshCells({ rowNodes: [node], force: true });
  };
  return (
    <button type="submit" onClick={markBookMark} className="bookmark-post">
      <svg
        width="24"
        height="24"
        xmlns="http://www.w3.org/2000/svg"
        style={{ fill: "#3b4252" }}
      >
        {data.bookMark ? (
          <path d="M18 24l-6-5.269-6 5.269v-24h12v24z" />
        ) : (
          <path d="M5 0v24l7-6 7 6v-24h-14zm1 1h12v20.827l-6-5.144-6 5.144v-20.827z" />
        )}
      </svg>
    </button>
  );
};

export default BookMarkPost;
