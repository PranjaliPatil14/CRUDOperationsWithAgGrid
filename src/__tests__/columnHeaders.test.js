import BookMarkPost from "../BookMarkPost";
import columnData from "../columnHeaders";
import DeletePost from "../DeletePost";

describe("columnHeaders", () => {
  test("should return all columns", () => {
    expect(columnData).toMatchSnapshot();
  });

  test("should return toLocaleDateString value for date column when date present in the row data", () => {
    const params = {
      data: {
        date: new Date("2020-05-12"),
      },
    };
    expect(columnData[0].valueGetter(params)).toEqual("5/12/2020");
  });

  test("should return value for date column when date is not present in the row data", () => {
    const params = {
      data: {
        id: 10,
      },
    };

    const expectedDate = new Date();
    expectedDate.setDate(10);

    expect(columnData[0].valueGetter(params)).toEqual(
      expectedDate.toLocaleDateString()
    );
  });

  test("should return Cell renderer framework for BookMark column", () => {
    expect(columnData[3].cellRendererFramework).toEqual(BookMarkPost);
  });

  test("should return Cell renderer framework for delete column", () => {
    expect(columnData[4].cellRendererFramework).toEqual(DeletePost);
  });
});
