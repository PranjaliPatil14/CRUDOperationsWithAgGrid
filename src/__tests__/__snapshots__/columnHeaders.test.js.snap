// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`columnHeaders should return all columns 1`] = `
Array [
  Object {
    "editable": false,
    "field": "date",
    "filter": "agDateColumnFilter",
    "filterParams": Object {
      "clearButton": true,
      "comparator": [Function],
    },
    "headerName": "Date",
    "minWidth": 80,
    "valueGetter": [Function],
  },
  Object {
    "cellClass": "bold-text",
    "cellEditorFramework": Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    },
    "field": "title",
    "headerName": "Post Title",
    "minWidth": 350,
  },
  Object {
    "field": "body",
    "headerName": "Post Content",
  },
  Object {
    "cellRendererFramework": [Function],
    "editable": false,
    "field": "bookMark",
    "filter": false,
    "headerName": "BookMark",
    "minWidth": 100,
    "sort": "desc",
  },
  Object {
    "cellRendererFramework": [Function],
    "editable": false,
    "filter": false,
    "headerName": "",
    "sortable": false,
    "width": 70,
  },
]
`;
